lib/src/Base/Func/FieldToFieldConnection.cxx

```diff
@@ -0,0 +1,205 @@
+//                                               -*- C++ -*-
+/**
+ * @brief The class that implements the composition between field functions or 
+ *        point to field and field to point functions
+ *
+ * Copyright 2005-2017 Airbus-EDF-IMACS-Phimeca
+ *
+ * Permission to copy, use, modify, sell and distribute this software
+ * is granted provided this copyright notice appears in all copies.
+ * This software is provided "as is" without express or implied
+ * warranty, and with no claim as to its suitability for any purpose.
+ *
+ *
+ */
+
+#include "openturns/FieldToFieldConnection.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(FieldToFieldConnection);
+
+static const Factory<FieldToFieldConnection> Factory_FieldToFieldConnection;
+
+/* Default constructor */
+FieldToFieldConnection::FieldToFieldConnection()
+  : FieldFunctionImplementation()
+  , isFieldFunctionComposition_(true)
+  , leftFieldFunction_()
+  , rightFieldFunction_()
+  , fieldToPointFunction_()
+  , pointToFieldFunction_()
```

Comment:
warning: field 'fieldToPointFunction_' will be initialized after field 'pointToFieldFunction_' [-Wreorder]

Reply:
@jschueller I changed the order in the header: it was more consistent to start with f and end with g when defining fog!